src/plugin/transform.ts

```diff
@@ -69,56 +67,39 @@ const registerTransform = (
     md: MarkdownIt,
     {embeddingMode, runtimeJsPath, output, bundle, updateTokens}: RegisterTransformOptions,
 ) => {
-    const directiveHandler: DirectiveBlockHandler = ({state, content}) => {
-        if (!content || !state) {
+    md.use(directiveParser());
+
+    registerContainerDirective(md, 'html', (state, params) => {
+        if (!params.content) {
             return false;
         }
 
-        const {env} = state;
-
-        addHiddenProperty(env, 'bundled', new Set<string>());
-
         if (updateTokens) {
             const tokenType = embeddingModeToTokenType[embeddingMode];
-
             const token = state.push(tokenType, TAG, 0);
-
-            token.block = true;
-            token.content = content;
+            // set fields like in fence token
+            token.map = [params.startLine, params.endLine];
+            token.content = params.content.raw;
+            token.markup = ':::';
+            token.info = 'html';
```

Comment:
Seems like this would be better handled by the (forked) directive plugin itself. I find it hard to believe that every single future plugin that would use `@diplodoc/directive` would choose to honor the convention as outlined here. To clarify, I feel like manually setting `token.markup` and `token.info` every time we'd like to transform a directive token into something we can render is a bit too verbose.

Reply:
You right.

I have implemented [helper for this case] in directive extension, so now html-directive can be registered as follows:

```js
md.use(directiveParser());

registerContainerDirective(md, {
    name: 'html',
    type: 'code_block',
    container: {
        tag: TAG,
        token: embeddingModeToTokenType[embeddingMode],
    },
    match: (_params, state) => {
        const {env} = state;
        env.meta ||= {};
        env.meta.script ||= [];
        env.meta.style ||= [];
        if (!env.meta.script.includes(runtimeJsPath)) {
            env.meta.script.push(runtimeJsPath);
        }

        addHiddenProperty(env, 'bundled', new Set<string>());

        if (bundle) {
            copyRuntimeFiles({runtimeJsPath, output}, env.bundled);
        }

        return true;
    },
});
```